sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/caches/AsyncCache.java

```diff
@@ -74,15 +75,15 @@ public void set(TKey key, TValue value) {
                 }
 
                 logger.debug("cache[{}] result value is obsolete ({}), computing new value", key, obsoleteValue);
-                AsyncLazy<TValue> asyncLazy = new AsyncLazy<>(singleValueInitFunc);
+                AsyncLazy<TValue> asyncLazy = new AsyncLazy<>(singleValueInitFunc, Optional.of(value));
```

Comment:
The api contract should provide hint whether it is ok to keep the old value or not!
add a new param `keepOldValue` to the method.